src/components/utils/useListNavigation/useListNavigation.ts

```diff
@@ -0,0 +1,179 @@
+import {RefObject, useCallback, useEffect, useLayoutEffect, useState} from 'react';
+
+import {moveBack} from './moveBack';
+import {moveForward} from './moveForward';
+
+export type UseListNavigationProps<ItemType, AnchorType> = {
+    items: ItemType[];
```

Comment:
If list item disabled, or become disabled it should be skipped with keyboard navigation

Reply:
Yes, `skip` property is responsible for that